src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java

```diff
@@ -303,6 +306,46 @@ public static JsonArrayBuilder json(List<DataverseContact> dataverseContacts) {
         }
         return jsonArrayOfContacts;
     }
+    
+    public static JsonArrayBuilder getOwnersFromDvObject(DvObject dvObject) {
+        
+        List <DvObject> ownerList = new ArrayList();
+        dvObject = dvObject.getOwner(); // We're going to ignore the object itself
+        while (dvObject != null) {
+            ownerList.add(dvObject);
+            dvObject = dvObject.getOwner();
+        } 
+        
+        JsonArrayBuilder jsonArrayOfOwners = Json.createArrayBuilder();
+        
+        for (DvObject dvo : ownerList){
+                JsonObjectBuilder ownerObject = jsonObjectBuilder();
+                if (dvo.isInstanceofDataverse()){
+                    ownerObject.add("type", "DATAVERSE");
+                }
+                if (dvo.isInstanceofDataset()){
+                    ownerObject.add("type", "DATASET");
+                }
+                 if (dvo.isInstanceofDataFile()){
+                    ownerObject.add("type", "DATAFILE");
+                }
+                if (dvo.isInstanceofDataverse()){
+                    Dataverse in = (Dataverse) dvo;
+                    ownerObject.add("identifier", in.getAlias());
```

Comment:
It would be beneficial to include the version as well to the ownerObject, because for a file, its owner might be the dataset doi:10.5072/FK2/WTBMGC with version=2.0, but its parent is not the same dataset at an earlier version such as 1.0. I hope I'm making myself clear

Reply:
I added the logic for including a version number, but since we're omitting the base object from the isPartOf it can only apply to Files, however there's no endpoint for getting a file by its version like there is for dataset - the current result is that if I use the getFileData (@Path("{id}")) it gets me the latest available version

Reply:
@MellyGray @GPortas I think I need https://github.com/IQSS/dataverse/issues/10280 to be completed before I can finish this the way it needs to be finished.

Reply:
Sure. I can do that. The reason I had put them together is that the url for the dataset page currently looks like this: http://localhost:8080/dataset.xhtml?persistentId=doi:10.5072/FK2/UE0DKV&version=2.0

Reply:
Got it, @sekmiller! I see where you're coming from, but we're aiming to follow an API-first approach to decouple the API implementation from the UI use cases, ensuring our API remains flexible for various applications. We don't know who is going to consume this endpoint, so we should keep these two pieces of information separate to allow them to be used independently. Thanks for implementing the suggestion